readme.md

```diff
@@ -1,161 +1,112 @@
-	match := SubscribeTestMyThingRegex.FindStringSubmatch(ch)
-	if len(match) < 2 {
-		return errors.New("channel did not match expected format: " + ch)
-	}
-
-	// Map the struct fields to the order they will appear in the topic
-	fields := []*string{&params.StreetlightId,&params.Action}
-
-	for i, field := range fields {
-		// Populate params fields - skipping the first index of 'match' as 
-		// captured groups start at index=1
-		*field = match[i+1]
-	}
-
-	return nil
-}
-
-// Build uses TurnOnParams to build the channel required for TurnOn operations
-func (params TurnOnParams) Build() string {
-	const ch string = "smartylighting/streetlights/1/0/action/{streetlightId}/turn/on"
-	r := strings.NewReplacer("{streetlightId}",params.StreetlightId)
-
-	return r.Replace(ch)
-}
+##### Pre-requisites
+To run the generated code the following needs to be installed
+
+1. go 1.16 +
+2. rabbitmq-server OR docker
 
+##### Running the code
+
+1. Navigate to the path where the code was generated
+2. Run the following commands to download the dependencies
+```bash
+go mod download
+go mod tidy
+```
+3. Currently the code does not utilize the server bindings to generate the server URI. It is currently hardcoded to point to a local instance of `rabbitmq`. It is hardcoded as `"amqp://guest:guest@localhost:5672/"` at `<generated-code>/config/server.go`. Change it as per your rabbitmq instance requirements
+4. Finally to execute the code run
+```bash
+go run main.go
 ```
-#### Controller Sample
-```golang
-//   Locations: controller.go
-
-// TurnOff implements operation.Producer
-func (c Controller) TurnOff(params channel.TurnOffParams, msg message.TurnOnOff) error {
-	// Define any operation bindings. These are constant per operation
-    var mqtt5Bindings = map[string]interface{} {"qos":1,"retain":true,"bindingVersion": "0.0.1",}
-
-	// Throw error for missing content type encoder
-	w := c.getWriter(msg.ContentType)
-	if w == nil {
-		return errors.New("no message writer is registered for content type: " + msg.ContentType)
-	}
-
-	// Throw error if failed to encode payload
-	if msg.RawPayload, err := w.Write(msg.Payload) {
-		return err
-	}
-
-	// Publish the underlying transport.Message with the transport layer
-	return c.async.Publish(params.Build(), mag.Message, kafkaBindings)
-}
-
-```
\ No newline at end of file
+5. Running local instance of `rabbitmq`, navigate to it using `http://localhost:15672/` with username and password `guest`/ `guest` (These are default rabbitmq credentials). 
+   FYI one can start an instance of `rabbitmq` using  `docker` as follow
+   ```
+   docker run -d -p 15672:15672 -p 5672:5672 rabbitmq:3-management
+   ```
+6. Create a queue as per the AsyncAPI spec. 
+   This can be done either of the following ways
+   a. Using the UI: Refer to this [article](https://www.cloudamqp.com/blog/part3-rabbitmq-for-beginners_the-management-interface.html) that walks through the process of how this can be done in the UI / RabbitMQ Admin 
```

Comment:
This multilevel list won't render properly. Rather you should use something like:

```
1. One
   - One.1
   - One.2
2. Two
   - Two.1
   - Two.2
3. Three
   - Three.1
   - Three.2
```

Reply:
done makes sense